### PR TITLE
Improve warnings

### PR DESCRIPTION
**Pull request by jnikula**
Our error handling does not seem to be very nice if the file or symbol is not found. Improve them.

### CODE REVIEW

File: src/hawkmoth/__init__.py [hunk 1]
```diff
@@ -167,7 +167,14 @@ def _get_filenames(self):
                                 location=(self.env.docname, self.lineno))
             return []
 
-        return [os.path.abspath(os.path.join(self.env.config.hawkmoth_root, filename))]
+        filename = os.path.abspath(os.path.join(self.env.config.hawkmoth_root, filename))
+
+        # Warn if the file is not found, but pass it to clang anyway
```
Comment (BrunoMSantos) [inline src/hawkmoth/__init__.py:184]: Hmm, could we let it fail during the parse instead and, in the `except`, figure out the specific cause and adjust the error message?

File: src/hawkmoth/__init__.py [hunk 2]
```diff
@@ -93,15 +93,27 @@ def __parse(self, filename):
...
+        if not match:
+            if self._get_names():
+                args = ' '.join(self.arguments)
+                self.logger.warning(f'"{self.name}:: {args}" does not match documented symbols.',
+                                    location=(self.env.docname, self.lineno))
+            else:
+                # autodoc
+                self.logger.warning('No documented symbols were found.',
+                                    location=(self.env.docname, self.lineno))
+
```
Comment (BrunoMSantos) [inline src/hawkmoth/__init__.py:116]: Nice one!

File: src/hawkmoth/__init__.py (hunk 1 above)
Comment (jnikula) [reply src/hawkmoth/__init__.py:184]: Can do.

Comment (jnikula) [reply src/hawkmoth/__init__.py]: Force pushed an update

File: src/hawkmoth/parser.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
 """
```
Comment (BrunoMSantos) [inline src/hawkmoth/parser.py:31]: Typo in commit message: 'fro' -> 'from'.

Thanks for the fix! Looks good to me.

Comment (jnikula) [reply src/hawkmoth/parser.py:31]: Thanks, fixed!

**Pull request merged**